### DISCUSSION TITLE
How to create expo-module without creating new project

### DISCUSSION THREAD

**Discussion by damianfilipek81** [expo/expo#26051] [Archived: Questions] [answered]
Hi guys, is there somewhere a documentation about how to create a expo-module inside existing project WITHOUT creating it as a separate-standalone project? We have a situation where we use react native as a module inside native app and i don't want to create separate project everytime i do some communication between native<=>javascript

**Comment by tsapeta** [comment, marked as answer]
I guess you're looking for local Expo modules, you can find more details here: https://docs.expo.dev/modules/get-started/#adding-a-new-module-to-an-existing-application
With the `--local` flag, it creates the module packages inside the `modules` folder within your existing app.

**Comment by dennisofficial** [comment]
i tried doing this, but when I compile it keeps telling me `cannot resolve module name "NAME"`

**Reply by MetalRoz** [reply]
could you resolve this problem dude?
